### PR TITLE
chore(docs): Add `@since` and `@history` annotations for Array

### PR DESCRIPTION
**Pull request by phated**
This updates the Array module doc blocks with `@since` and `@history` attributes. I also noticed that I missed 1 `@returns` in my initial PR.

I also checked in the generated `array.md` file.

### CODE REVIEW

File: stdlib/array.md [hunk 1]
```diff
@@ -0,0 +1,1008 @@
...
+
+<details>
+<summary>Added in <code>0.2.0</code></summary>
+<table>
+<thead>
+<tr><th>version</th><th>changes</th></tr>
+</thead>
+<tbody>
+<tr><td><code>0.1.0</code></td><td>Originally named `arrays`</td></tr>
+<tr><td><code>0.2.0</code></td><td>Renamed to `array`</td></tr>
```
Comment (phated) [inline stdlib/array.md:15]: So, there's technically edge-case bug with these: markdown inside html isn't processed as markdown. I'm not sure the best way to handle (I really don't want to parse and convert it in graindoc).

File: stdlib/array.gr [hunk 2]
```diff
@@ -32,6 +37,9 @@ let initLength = (length) => {
  * Returns the length of the input array.
  *
  * @param array: The array to inspect
+ * @returns The amount of elements in the array
```
Comment (ospencer) [inline stdlib/array.gr:40]:
```suggestion
 * @returns The number of elements in the array
```

**Review by ospencer — COMMENTED** [review]
Looks good! One small tweak but otherwise 👍 from me.

File: stdlib/array.md (hunk 1 above)
Comment (peblair) [reply stdlib/array.md:15]: Deferring to @ospencer on what the best option would be here

Comment (phated) [reply stdlib/array.md:15]: I talked with @ospencer and he said we should just leave this for now.

**Review by ospencer — APPROVED** [review]

**Review by peblair — APPROVED** [review]

**Review by marcusroberts — APPROVED** [review]

**Pull request merged**